File: c#/Book/PetrolPump/PetrolPump/PetrolPump/Pump/FillState.cs
namespace PetrolPump.Pump
{
  public class FillState 
  {
    public enum Type 
    {
        IDLE,
        FILLING,
        SALE_COMPLETE
    }

    public FillState(Type mode) 
    {
        this.Mode = mode;
        this.Sale = null;
    }

    public FillState(Type mode, Sale sale) 
    {
        this.Mode = mode;
        this.Sale = sale;
    }

    public readonly Type Mode;
    public readonly Sale Sale;
}

}

File: book/petrol-pump/java/pump/FillState.java
package pump;

public class FillState {
    public enum Type {
        IDLE,
        FILLING,
        SALE_COMPLETE
    }

    public FillState(Type mode) {
        this.mode = mode;
        this.sale = null;
    }

    public FillState(Type mode, Sale sale) {
        this.mode = mode;
        this.sale = sale;
    }

    public final Type mode;
    public final Sale sale;
}

